carbon-design-system/carbon-addons-iot-react
# feat(table): add support for size prop to effect header/aggregation footer

Closes #1986 

**Depends on: #2811**

Since this addition basically touched all the stories, I based it on the table cleanup stories PR to prevent a ton of merge conflicts.

**Summary**
- Previously the carbon `size` prop would only effect rows within the table, but now it will also change the size of the aggregation row, or the header. The header will only shrink as small as it's able, so if you have an overflow menu, or sortable columns then it will remain roughly the same default size; however, it will grow larger if you pass the `xl` option.

**Change List (commits, features, bugs, etc)**

- Added `size` prop to table
- updated styles to allow various sizes to adjust the header/aggregation footer height properly
- added `size` knob to all stories

**Acceptance Test (how to verify the PR)**

- check all the table/stateful table stories changing the size knob on each and confirm the header/rows/aggregation footer all update accordingly. Note the header may not shrink smaller if it has sorting or an overflow menu.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Confirm no adverse visual side-affects in the stories



**Things to look for during review**
